src/Core_Fix_StudioOptimizations/FixGuideObject.cs

```diff
@@ -0,0 +1,21 @@
+﻿using BepInEx;
+using HarmonyLib;
+using Studio;
+using UnityEngine;
+
+namespace IllusionFixes
+{
+    public partial class StudioOptimizations : BaseUnityPlugin
+    {
+        [HarmonyPrefix]
+        [HarmonyPatch(typeof(Studio.GuideObject), nameof(Studio.GuideObject.CalcRotation))]
+        private static void GuideObjectCalcRotationPrefix(Studio.GuideObject __instance)
+        {
+            if( !__instance.m_Enables[1] && __instance.transformTarget != null && __instance.mode == GuideObject.Mode.LocalIK )

```

Comment:
> Is m_Enables always guaranteed to have at least 2 items in it?  

It's okay, because we see a lot of codes that assume that there are more than three elements.

![image](https://github.com/IllusionMods/IllusionFixes/assets/4230203/b1d54504-f8f1-47b0-9bde-7e52dd6d1fab)
![image](https://github.com/IllusionMods/IllusionFixes/assets/4230203/50c5414b-127c-49c9-a9ba-d0a8a7b2a93e)
![image](https://github.com/IllusionMods/IllusionFixes/assets/4230203/cdc68073-e97c-49ee-bf8e-60468fa305ac)

> Checking mode first might be faster.

Indeed.
I thought, but I checked.
Moved to the top.
I'm wondering which is more expensive, the 9% difference or the array reference cost.


```
        static uint total;
        static uint condition0;
        static uint condition1;
        static uint condition2;

        [HarmonyPrefix]
        [HarmonyPatch(typeof(Studio.GuideObject), nameof(Studio.GuideObject.CalcRotation))]
        private static void GuideObjectCalcRotationPrefix(Studio.GuideObject __instance)
        {
            if (!__instance.m_Enables[1]) ++condition0;
            if (__instance.transformTarget != null) ++condition1;
            if (__instance.mode == GuideObject.Mode.LocalIK) ++condition2;
            ++total;

            if( total % 10 == 0 )
            {
                System.Console.WriteLine($"@@ {(float)condition0/total:P},{(float)condition1 / total:P},{(float)condition2 / total:P}");
            }

            if ( !__instance.m_Enables[1] && __instance.transformTarget != null && __instance.mode == GuideObject.Mode.LocalIK )
            {
                //Initialize rotation so that IKs with disabled rotation follow the parent's rotation
                __instance.transformTarget.localRotation = Quaternion.identity;
            }
        }
```

```
@@ 31.25%,100.00%,40.63%
```